drivers/gpio/gpio_sx1509b.c

```diff
@@ -650,6 +694,71 @@ static const struct gpio_driver_api api_table = {
 #endif
 };
 
+int sx1509b_pwm_pin_configure(struct device *dev, gpio_pin_t pin)
+{
+	const struct sx1509b_config *cfg = dev->config_info;
+	struct sx1509b_drv_data *drv_data = dev->driver_data;
+	struct sx1509b_pin_state *pins = &drv_data->pin_state;
+	int rc = 0;
+
+	/* Can't do I2C bus operations from an ISR */
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+
+	if (pin >= ARRAY_SIZE(intensity_registers)) {
+		return -ERANGE;
+	}
+
+	k_sem_take(&drv_data->lock, K_FOREVER);
+
+	/* Enable LED driver */
+	drv_data->led_drv_enable |= BIT(pin);
+	rc |= i2c_reg_write_word_be(drv_data->i2c_master, cfg->i2c_slave_addr,
```

Comment:
This idiom isn't right.  Please follow the rest of the Zephyr codebase in using straight assignment to `rc`, and only proceeding with subsequent operations if `rc == 0` (or `rc >= 0`, depending on taste).